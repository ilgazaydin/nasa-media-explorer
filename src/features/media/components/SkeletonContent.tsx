/**
 * @file SkeletonContent.tsx
 * @description A placeholder skeleton component for the Media Detail page content.
 *
 * @component
 * @returns {JSX.Element} A MUI `Paper` containing skeleton elements to simulate a large preview,
 * heading, metadata, and description while content is being fetched.
 *
 * @usage
 * Shown while the Media Detail page is loading data such as metadata and asset previews.
 *
 * @example
 * {isLoading && <SkeletonContent />}
 */

import { Box, Paper, Skeleton } from "@mui/material";

const SkeletonContent = () => (
  <Paper sx={{ p: 3, borderRadius: 2 }}>
    <Box>
      <Skeleton
        data-testid="skeleton-line"
        variant="rectangular"
        height={400}
        sx={{ mb: 3 }}
      />
      <Skeleton
        data-testid="skeleton-line"
        variant="text"
        height={40}
        width="70%"
      />
      <Skeleton
        data-testid="skeleton-line"
        variant="text"
        height={24}
        width="40%"
      />
      <Skeleton
        data-testid="skeleton-line"
        variant="text"
        height={20}
        width="90%"
        sx={{ mt: 2 }}
      />
      <Skeleton
        data-testid="skeleton-line"
        variant="text"
        height={20}
        width="90%"
      />
      <Skeleton
        data-testid="skeleton-line"
        variant="text"
        height={20}
        width="70%"
      />
    </Box>
  </Paper>
);

export default SkeletonContent;
