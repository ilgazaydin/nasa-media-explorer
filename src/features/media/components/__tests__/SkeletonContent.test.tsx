import { render, screen } from "@testing-library/react";
import SkeletonContent from "../SkeletonContent";

describe("SkeletonContent", () => {
  it("renders 6 skeleton elements representing media detail loading layout", () => {
    render(<SkeletonContent />);
    const skeletons = screen.getAllByTestId("skeleton-line");
    expect(skeletons).toHaveLength(6);
  });

  it("is wrapped in a MUI Paper container", () => {
    const { container } = render(<SkeletonContent />);
    const paper = container.querySelector(".MuiPaper-root");
    expect(paper).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<SkeletonContent />);
    expect(asFragment()).toMatchSnapshot();
  });
});
