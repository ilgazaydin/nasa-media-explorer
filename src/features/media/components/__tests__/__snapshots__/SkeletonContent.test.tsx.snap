// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SkeletonContent > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-9eitor-MuiPaper-root"
    style="--Paper-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiBox-root css-0"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-6pusbi-MuiSkeleton-root"
        data-testid="skeleton-line"
        style="height: 400px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-kkcbhd-MuiSkeleton-root"
        data-testid="skeleton-line"
        style="width: 70%; height: 40px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-kkcbhd-MuiSkeleton-root"
        data-testid="skeleton-line"
        style="width: 40%; height: 24px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-1icoqhh-MuiSkeleton-root"
        data-testid="skeleton-line"
        style="width: 90%; height: 20px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-kkcbhd-MuiSkeleton-root"
        data-testid="skeleton-line"
        style="width: 90%; height: 20px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-kkcbhd-MuiSkeleton-root"
        data-testid="skeleton-line"
        style="width: 70%; height: 20px;"
      />
    </div>
  </div>
</DocumentFragment>
`;
